pkg/server/server.go

```diff
@@ -497,7 +497,7 @@ func (s *Server) Run(ctx context.Context) error {
 	}
 
 	if s.options.Virtual.Enabled {
-		if err := s.installVirtualWorkspaces(ctx, server, kubeClusterClient, dynamicClusterClient, kcpClusterClient, genericConfig.Authentication, genericConfig.ExternalAddress, preHandlerChainMux); err != nil {
+		if err := s.installVirtualWorkspaces(ctx, genericConfig.LoopbackClientConfig, server, kubeClusterClient, dynamicClusterClient, kcpClusterClient, genericConfig.Authentication, genericConfig.ExternalAddress, preHandlerChainMux); err != nil {
```

Comment:
use controllerConfig.

Also we should not pass config and clients. All the clients should be instantiated in installVirtualWorkspaces, with the right user-agent, as we do it with controllers.